src/lib/icon/icon.ts

```diff
@@ -162,6 +170,12 @@ export class MatIcon extends _MatIconMixinBase implements OnChanges, OnInit, Can
   private _previousFontSetClass: string;
   private _previousFontIconClass: string;
 
+  /** Keeps track of the current page path. */
+  private _previousPath?: string;
+
+  /** Keeps track of the elements and attributes that we've prefixed with the current path. */
+  private _elementsWithExternalReferences?: Map<Element, {name: string, value: string}[]>;
```

Comment:
It looks like you reuse this as `cachedElements` in the `ngAfterViewChecked` method so that its a shorter name, should we just use `cachedElements` as the property name?